executor/utils/workdir_resolver.py

```
@@ -0,0 +1,197 @@
+    2) bot-level cwd (validated)
+    3) managed workspace root by task_id (fallback for invalid explicit inputs)
+    """
+    resolution = resolve_task_workdir_details(task_data, bot_cwd)
+    if not resolution.effective_cwd:
+        return None
+    # Ensure directory exists (caller expects a usable cwd).
+    # In Docker mode, the workspace root is expected to exist in the container,
+    # but creating the leaf directory is still safe.
+    try:
+        os.makedirs(resolution.effective_cwd, exist_ok=True)
+    except Exception as exc:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)